### DISCUSSION TITLE
languages: what is responsible for auto-indenting new `{}` blocks?

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [zed-industries/zed#10970] [Feature Requests]
Most languages will automatically insert a newline and tab when pressing `Enter` while inbetween `{` `}`.
TypeScript for example:
![image](https://github.com/zed-industries/zed/assets/158013953/c785d765-b52e-4e89-a222-509c4b0f9de6)

I'm trying to get this working for Kotlin detailed at https://github.com/zed-extensions/zed-kotlin/issues/5. I have newline insertion working, but can't figure out which config file controls the tab insertion. or if an entire language server protocol / formatter needs to be implemented for that behavior. Anyone know?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
